packages/docusaurus-theme-classic/src/theme/Navbar/index.js

```diff
@@ -35,6 +35,17 @@ function NavLink({to, href, label, position, ...props}) {
         : {
             activeClassName: 'navbar__link--active',
             to: toUrl,
+            isActive: (match, location) => {
+              const firstPathSegment = toUrl
+                .replace(/^\//, '')
+                .split('/')
+                .shift();
+              const isDocPage =
```

Comment:
Not great to have doc-specific code in Navbar :/

Reply:
It depends. How else to fix it? I do not here anything bad, considering that the base route for docs here will also change if it will be edited into the plugin settings.